azurerm/internal/services/synapse/synapse_private_link_hub.go

```diff
@@ -0,0 +1,167 @@
+		}),
+
+		Schema: map[string]*pluginsdk.Schema{
+			"name": {
+				Type:         pluginsdk.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validate.PrivateLinkHubName,
+			},
+
+			"resource_group_name": azure.SchemaResourceGroupName(),
+
+			"location": azure.SchemaLocation(),
+
+			"tags": tags.Schema(),
+		},
+	}
+}
+
+func resourceSynapsePrivateLinkHubCreate(d *pluginsdk.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).Synapse.PrivateLinkHubsClient
+	ctx, cancel := timeouts.ForCreate(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	name := d.Get("name").(string)
+	resourceGroup := d.Get("resource_group_name").(string)
+
+	existing, err := client.Get(ctx, resourceGroup, name)
+	if err != nil {
+		if !utils.ResponseWasNotFound(existing.Response) {
+			return fmt.Errorf("checking for present of existing Synapse Private Link Hub %q (Resource Group %q): %+v", name, resourceGroup, err)
+		}
+	}
+	if existing.ID != nil && *existing.ID != "" {
+		return tf.ImportAsExistsError("azurerm_synapse_private_link_hub", *existing.ID)
+	}
+
+	privateLinkHubInfo := synapse.PrivateLinkHub{
+		Location: utils.String(location.Normalize(d.Get("location").(string))),
+		Tags:     tags.Expand(d.Get("tags").(map[string]interface{})),
+	}
+
+	_, err = client.CreateOrUpdate(ctx, privateLinkHubInfo, resourceGroup, name)
+	if err != nil {
+		return fmt.Errorf("creating Synapse Private Link Hub %q (Resource Group %q): %+v", name, resourceGroup, err)
+	}
+
+	subscriptionId := meta.(*clients.Client).Account.SubscriptionId
+	id := parse.NewPrivateLinkHubID(subscriptionId, resourceGroup, name)
+	d.SetId(id.ID())
+
+	return resourceSynapsePrivateLinkHubRead(d, meta)
+}
+
+func resourceSynapsePrivateLinkHubRead(d *pluginsdk.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).Synapse.PrivateLinkHubsClient
+	ctx, cancel := timeouts.ForRead(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	id, err := parse.PrivateLinkHubID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	resp, err := client.Get(ctx, id.ResourceGroup, id.Name)
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			log.Printf("[INFO] synapse %q does not exist - removing from state", d.Id())
+			d.SetId("")
+			return nil
+		}
+		return fmt.Errorf("retrieving Synapse Private Link Hub %q (Resource Group %q): %+v", id.Name, id.ResourceGroup, err)
+	}
+
+	d.Set("name", id.Name)
+	d.Set("resource_group_name", id.ResourceGroup)
+	d.Set("location", location.NormalizeNilable(resp.Location))
+
+	return tags.FlattenAndSet(d, resp.Tags)
+}
+
+func resourceSynapsePrivateLinkHubUpdate(d *pluginsdk.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).Synapse.PrivateLinkHubsClient
+	ctx, cancel := timeouts.ForUpdate(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	id, err := parse.PrivateLinkHubID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	if d.HasChanges("tags") {
```

Comment:
Can use d.HasChange if we're only checking one property
```suggestion
	if d.HasChange("tags") {
```